os_migrate/roles/import_workloads/tasks/workload.yml

```diff
@@ -1,133 +1,135 @@
 - name: import a workload
   block:
-  - name: preliminary setup for workload import
-    os_migrate.os_migrate.import_workload_prelim:
-      cloud: dst
-      validate_certs: "{{ os_migrate_dst_validate_certs|default(omit) }}"
-      ca_cert: "{{ os_migrate_dst_ca_cert|default(omit) }}"
-      client_cert: "{{ os_migrate_dst_client_cert|default(omit) }}"
-      client_key: "{{ os_migrate_dst_client_key|default(omit) }}"
-      dst_filters: "{{ os_migrate_dst_filters }}"
-      src_conversion_host:
-        "{{ os_src_conversion_host_info.openstack_conversion_host }}"
-      log_dir: "{{ os_migrate_data_dir }}/workload_logs"
-      data: "{{ item }}"
-    register: prelim
+    - name: preliminary setup for workload import
+      os_migrate.os_migrate.import_workload_prelim:
+        cloud: dst
+        validate_certs: "{{ os_migrate_dst_validate_certs|default(omit) }}"
+        ca_cert: "{{ os_migrate_dst_ca_cert|default(omit) }}"
+        client_cert: "{{ os_migrate_dst_client_cert|default(omit) }}"
+        client_key: "{{ os_migrate_dst_client_key|default(omit) }}"
+        dst_filters: "{{ os_migrate_dst_filters }}"
+        src_conversion_host:
+          "{{ os_src_conversion_host_info.openstack_conversion_host }}"
+        log_dir: "{{ os_migrate_data_dir }}/workload_logs"
+        data: "{{ item }}"
+      register: prelim
+      when: data.data_copy|default(true)
 
-  - name: show progress data
-    ansible.builtin.debug:
-      msg:
-        - "{{ prelim.server_name }} log file: {{ prelim.log_file }}"
-        - "{{ prelim.server_name }} progress file: {{ prelim.state_file }}"
-    when: prelim.changed
+    - name: show progress data
+      ansible.builtin.debug:
+        msg:
+          - "{{ prelim.server_name }} log file: {{ prelim.log_file }}"
+          - "{{ prelim.server_name }} progress file: {{ prelim.state_file }}"
+      when: prelim.changed and data.data_copy|default(true)
```

Comment:
`prelim` may be undefined if it wasn't executed. So we should probably do something like:

```
when:
  - prelim is defined and prelim.changed
  - data.data_copy|default(true)
```

This "is defined" guard on usage of prelim will have to be done in the `prelim` uses in the rest of the code below too.